File: app/models/card.rb
# == Schema Information
#
# Table name: cards
#
#  id          :integer          not null, primary key
#  list_id     :integer          not null
#  title       :string(255)      not null
#  description :text
#  due_date    :datetime
#  open        :boolean          default(TRUE), not null
#  created_at  :datetime         not null
#  updated_at  :datetime         not null
#  position    :integer
#

class Card < ActiveRecord::Base
  default_scope {order("cards.position")}
  acts_as_list

  acts_as_taggable_on :languages, :growth, :experience

  belongs_to :list
  belongs_to :assignee, class_name: "User"

  has_one :board, through: :list

  has_many :cards_members, class_name: "CardMember"
  has_many :members, through: :cards_members

  has_many :comments, class_name: "CardComment"
  has_many :activities, class_name: "CardActivities"

  accepts_nested_attributes_for :comments

end

File: frontend/app/models/card.js
import DS from 'ember-data';

export default DS.Model.extend({
  listId: DS.attr(),
  title: DS.attr('string'),
  description: DS.attr('string'),
  dueDate: DS.attr('date'),
  open: DS.attr('boolean'),
  commentCount: DS.attr(),
  url: DS.attr('string'),
  assigneeId: DS.attr(),
  position: DS.attr(),
  companyId: DS.attr(),
  salary: DS.attr(),
  createdAt: DS.attr("date"),
  updatedAt: DS.attr("date"),
  list: DS.belongsTo('list')
});
